weather@mockturtl/src/met_norway.ts

```diff
@@ -0,0 +1,674 @@
+        let resultStripped = (result > 100) ? result-100 : result;
+        let conditionIDStripped = (conditionID > 100) ? conditionID - 100 : conditionID; 
+        // Make the comparison, keep the polar night condition id
+        if (conditionIDStripped > resultStripped) result = conditionID;
+      }
+      // If there is no rain or worse, just get the most common condition for the day
+      if (result <= 4) {
+        return this.GetMostCommonCondition(conditions);
+      }
+      return conditions[result].name;
+    }
+
+    private ParseCurrentWeather(element: any, from: Date, to: Date): WeatherForecast {
+      return {
+        temperature: parseFloat(element.temperature['value']),
+        lat: element["latitude"],
+        lon: element["longitude"],
+        windDirection: parseFloat(element.windDirection["deg"]),
+        windSpeed: parseFloat(element.windSpeed["mps"]),
+        humidity: parseFloat(element.humidity["value"]),
+        pressure: parseFloat(element.pressure["value"]),
+        cloudiness: parseFloat(element.cloudiness["percent"]),
+        //fog: parseFloat(element.fog["@percent"]),
+        //lowClouds: parseFloat(element.lowClouds["percent"]),
+        //mediumClouds: parseFloat(element.mediumClouds["percent"]),
+        //highClouds: parseFloat(element.highClouds["percent"]),
+        //dewpointTemperature: parseFloat(element.dewpointTemperature["value"]),
+        from: from,
+        to: to
+      };
+    }
+
+    private Parse6HourForecast(element: any, from: Date, to: Date): SixHourForecast {
+      return {
+        //precipitation: parseFloat(element.precipitation["value"]),
+        minTemperature: parseFloat(element.minTemperature["value"]),
+        maxTemperature: parseFloat(element.maxTemperature["value"]),
+        from: from,
+        to: to,
+        altitude: parseInt(element.altitude),
+        symbol: element.symbol["id"],
+        symbolID: element.symbol["number"]
+      }
+    }
+
+    private ParseHourlyForecast(element: any, from: Date, to: Date): HourlyForecast {
+      return {
+        //precipitation: parseFloat(element.precipitation["value"]),
+        from: from,
+        to: to,
+        symbol: element.symbol["id"],
+        symbolID: element.symbol["number"],
+        altitude: parseInt(element.altitude)
+      }
+    }
+
+    private GetUrl(): string {
+        let location = this.app._location.replace(" ", "");
+        let url = this.baseUrl + "lat=";
+        if (!isCoordinate(location)) return "";
+
+        let latLon = location.split(",");
+        url += (latLon[0] + "&lon=" + latLon[1]);
+        return url;
+    }
+
+    private IsNight(): boolean {
+      if (!this.sunTimes) return false;
+      let now = new Date();
+      if (now < this.sunTimes.sunrise || now > this.sunTimes.sunset) return true;
+      return false;
+    }
+
+    private ResolveCondition(icon: string, checkIfNight?: boolean): Condition {
```

Comment:
I need help on writing text for conditions, it has much to be desired at the moment and I don't want to edit much of the text later because of translations.

I can't go out in this state in my opinion.

Reply:
I am struggling to find the code fragment you've highlighted in the files.  That question mark looks like a mangled ternary comparison, but as I don't use .ts I may be missing something.  When I expand the files looking for it a) I'm not finding it b) it's taking a lot of time not to find it c) what I am finding looks reasonable.

Reply:
The code above is shown but I wanted to select the code below, crap.

In typescript, for variables, paramaters and anything else you write types like this ```<varname>: <vartype>```, the ```?``` inditcates thet its an optional parameter.

Reply:
I meant the cases inside the function:
![image](https://user-images.githubusercontent.com/22377521/73250819-8f734580-41af-11ea-978d-4a144b37b02b.png)

(I'm unable to link it properly)

For each weather symbol (condition/icon, whatever) I'm need text what is not provided ```main and description fields```, so I have to write them based on the icon name, what is challenging because I don't know the intent behind some of them and English is not my first languge.

For example this one has a typo as well.

I contacted MET Norway for some explanation on the intent behind them, but they still need a rework.

Reply:
They don't answer.

I resolve it for now, and I'll make changes once I come up with something better.

I want the update out because some of the fixes are actually urgent.